src/components/forms/FormSearchField.js

```diff
@@ -1,46 +1,91 @@
+    options,
+    size = "small",
+    renderCustomOption,
+    handleSearch,
+    form: { setFieldValue, ...form },
+    valueKey,
+    labelKey,
     ...props
 }) => {
-    const errorMsg = getFormError(field.name, touched, errors);
-    const onOptionSelected = (option) => {
-        setFieldValue(field.name, option ? option[props.valueKey] : "");
-    };
-    const onSearchBlur = (event) => {
-        setFieldTouched(field.name, true);
+    const [searchValue, setSearchValue] = React.useState(value);
+    const filter = createFilterOptions();
+    const { t } = useTranslation("common");
+
+    React.useEffect(() => {
+        const val = {};
+        val[valueKey] = value;
+        setSearchValue(val);
+    }, [value, setSearchValue, valueKey]);
+
+    const onOptionSelected = (event, newValue) => {
+        if (newValue) {
+            if (typeof newValue === "string") {
+                setFieldValue(field.name, newValue);
+            } else if (newValue && newValue.inputValue) {
+                // Create a new value from the user input
+                setFieldValue(field.name, newValue.inputValue);
+            } else {
+                setFieldValue(field.name, newValue[valueKey]);
+            }
+        } else {
+            setFieldValue(field.name, "");
+        }
     };
 
     return (
-        <FormControl fullWidth error={!!errorMsg}>
-            <Autocomplete
-                label={label}
-                controlShouldRenderValue
-                isClearable={!required}
-                cacheOptions
-                defaultInputValue={value}
-                onChange={onOptionSelected}
-                onBlur={onSearchBlur}
-                {...field}
-                {...props}
-            />
-            <FormHelperText>{errorMsg || helperText}</FormHelperText>
-        </FormControl>
+        <Autocomplete
+            id="searchField"
+            getOptionSelected={(option, value) =>
+                option[labelKey] === value[labelKey]
+            }
+            getOptionLabel={(option) => (option && option[labelKey]) || ""}
+            options={options}
+            size={size}
+            freeSolo={true}
+            selectOnFocus={true}
+            value={searchValue}
+            handleHomeEndKeys={true}
+            onInputChange={handleSearch}
+            onChange={onOptionSelected}
+            renderOption={(option, state) => renderCustomOption(option)}
+            filterOptions={(options, params) => {
+                const filtered = filter(options, params);
+                const { inputValue } = params;
+                // Suggest the creation of a new value
+                if (params.inputValue !== "") {
+                    filtered.push({
+                        inputValue: params.inputValue,
+                        label: t("formatTermFreeTextOption", {
+                            inputValue: inputValue,
+                        }),
+                    });
+                }
```

Comment:
No need to reference `params.inputValue` anymore, and `inputValue: inputValue` is now redundant:
```suggestion
                const { inputValue } = params;
                // Suggest the creation of a new value
                if (inputValue) {
                    filtered.push({
                        inputValue,
                        label: t("formatTermFreeTextOption", { inputValue }),
                    });
                }
```